mlir/lib/Dialect/Linalg/Transforms/Vectorization.cpp

```diff
@@ -3000,13 +3018,21 @@ struct Conv1DGenerator
   /// kw is always unrolled.
   /// TODO: w (resp. kw) is unrolled when the strideW ( resp. dilationW) is
   /// > 1.
-  FailureOr<Operation *> depthwiseConv(bool flatten) {
+  FailureOr<Operation *> depthwiseConv(uint64_t channelDimVecSize,
+                                       bool flatten) {
     if (!valid)
       return rewriter.notifyMatchFailure(op, "unvectorizable depthwise conv");
 
+    bool scalableChDim = false;
     int64_t nSize, wSize, cSize, kwSize;
     // kernel{kw, c}
     bindShapeDims(rhsShapedType, kwSize, cSize);
+    // Dynamic channel size implies scalable vectorisation
+    if (ShapedType::isDynamic(cSize)) {
+      assert(channelDimVecSize != 0 && "Channel dim vec size must be > 0");
+      cSize = channelDimVecSize;
+      scalableChDim = true;
+    }
```

Comment:
Why a dynamic channel dimension implies scalable vectors? We have to make sure this also works for non scalable cases.

Reply:
I wonder this too (though I'm not very familiar with the linalg vectorizer). Could also use scalable vectors if your channel dimension was large (say 100 elements), but still a static size?

Would it be possible to control this with the by passing a the scalable dims flags for the vector sizes, like is the case with matmuls?

Reply:
> Why a dynamic channel dimension implies scalable vectors?

Because I run out of time this week an that's the case I need the most :) I will send an update shortly to add support for more generic case too. It's actually quite easy. The most labor-intensive task is writing tests for these convs.

> Could you also use scalable vectors if your channel dimension was large (say 100 elements), but still a static size?

Yes, we already do that for elementwise Ops. However, you need to tile first - that will lead to tensors with dynamic shapes.